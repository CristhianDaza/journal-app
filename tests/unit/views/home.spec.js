import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

describe('Pruebas en Home', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(Home)
  })

  test('debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('hacer click en el boton debe redireccionar a no-entry', () => {
    const mockRouter = {
      push: jest.fn()
    }
    const wrapper = shallowMount(Home, {
      global: {
        mocks: {
          $router: mockRouter
        }
      }
    })

    wrapper.find('button').trigger('click')

    expect(mockRouter.push).toHaveBeenCalled()
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
  })
})
