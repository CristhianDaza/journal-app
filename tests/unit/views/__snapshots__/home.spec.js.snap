// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home debe hacer match con el snapshot 1`] = `
<div class="home">
  <h1>Home</h1>
</div>
`;

exports[`Pruebas en Home debe hacer match con el snapshot 1`] = `
<div class="home">
  <h1>Hola Mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
